infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -96,6 +97,26 @@ private void applyDevfileVolumes(InternalMachineConfig machineConfig) {
     }
   }
 
+  private void applyDevfileEndpoints(InternalMachineConfig machineConfig)
+      throws InfrastructureException {
+    for (org.eclipse.che.api.core.model.workspace.devfile.Endpoint endpoint :
+        component.getEndpoints()) {
+      if (!machineConfig.getServers().containsKey(endpoint.getName())) {
+        machineConfig
+            .getServers()
+            .put(endpoint.getName(), ServerConfigImpl.createFromEndpoint(endpoint));
+      } else {
+        throw new InfrastructureException(
+            format(
+                "Devfile overrides the endpoint '%s' of the plugin/editor component '%s'. "
+                    + "This is not allowed because it would most probably cause the workspace "
```

Comment:
"cause the workspace **to** malfunction", but I might prefer `cause issues when running the workspace`

Reply:
I was wondering about that, too. Looked up "malfunction" in dictionary and found it could be used as a noun, too.

Reply:
Yeah, in that case I'd expect to see `cause a malfunction in the workspace` instead. Also, in my mind, malfunction has a hardware connotation, but that's maybe the most minor quibble I could have.

Reply:
fixed